lib/improver/spotdata/neighbour_finding.py

```diff
@@ -0,0 +1,445 @@
+        return function(cube, sites, **kwargs)
+
+    @staticmethod
+    def fast_nearest_neighbour(cube, sites, ancillary_data=None):
+        '''
+        Use iris coord.nearest_neighbour_index function to locate the nearest
+        grid point to the given latitude/longitude pair.
+
+        Performed on a 2D-surface; consider using the much slower
+        iris.analysis.trajectory.interpolate method for a more correct nearest
+        neighbour search with projection onto a spherical surface; this is
+        typically much slower.
+
+        Args:
+        -----
+        cube           : Iris cube of gridded data.
+        sites          : Dictionary of site data, including lat/lon and
+                         altitude information.
+                         e.g. {<site_id>: {'latitude': 50, 'longitude': 0,
+                                           'altitude': 10}}
+        ancillary_data : A dictionary containing additional model data that
+                         is needed. e.g. {'orography': <cube of orography>}
+
+        Returns:
+        --------
+        neighbours: Numpy array of grid i,j coordinates that are nearest to
+                    each site coordinate given. Includes height difference
+                    between site and returned grid point if orography is
+                    provided.
+
+        '''
+        if ancillary_data is not None and ancillary_data['orography']:
+            calculate_dz = True
+            orography = data_from_ancillary(ancillary_data, 'orography')
+        else:
+            calculate_dz = False
+
+        neighbours = np.empty(len(sites), dtype=[('i', 'i8'),
+                                                 ('j', 'i8'),
+                                                 ('dz', 'f8'),
+                                                 ('edge', 'bool_')])
+
+        # Check cube coords are lat/lon, else transform lookup coordinates.
+        trg_crs = xy_test(cube)
+
+        imax = cube.coord(axis='y').shape[0]
+        jmax = cube.coord(axis='x').shape[0]
+
+        for i_site, site in enumerate(sites.itervalues()):
+            latitude, longitude, altitude = (site['latitude'],
+                                             site['longitude'],
+                                             site['altitude'])
+
+            longitude, latitude = xy_transform(trg_crs, latitude, longitude)
+            i_latitude, j_longitude = get_nearest_coords(cube, latitude,
+                                                         longitude)
+
+            dz_site_grid = 0.
+            if calculate_dz:
+                dz_site_grid = altitude - orography[i_latitude, j_longitude]
+
+            neighbours[i_site] = (int(i_latitude), int(j_longitude),
+                                  dz_site_grid,
+                                  (i_latitude == imax or j_longitude == jmax))
+
+        return neighbours
+
+#     @staticmethod
+#     def nearest_neighbour(cube, sites, ancillary_data=None):
```

Comment:
I was waiting to see if AVD made their changes so I could include this, but yes, I'll exclude it from the pull request for now.